zen-quotes-app/src/app/home/home.component.ts

```diff
@@ -0,0 +1,65 @@
+import {Component, OnInit} from '@angular/core';
+import {QuotesService} from "../quotes.service";
+import {Quote} from "../../types/quote.type";
+
+@Component({
+  selector: 'app-home',
+  templateUrl: './home.component.html',
+  styleUrls: ['./home.component.scss']
+})
+
+export class HomeComponent implements OnInit {
+  title = "Inspiration";
+  quotes: Quote[] = [];
+  authors: string[] = [];
+  authorsFilter: string[] = [];
+  randomQuote: Quote = {author: "", text: ""};
+  quotesFiltered: Quote[] = [];
+  quotesPerPage: number = 9;
+  pagesArray: number[] = [];
+  activePage: number = 0;
+
+  constructor(private quotesService: QuotesService) {}
+
+  getAuthors() {
+    this.authors = [...new Set(this.quotes.map(quote => quote.author))].sort();
+  }
+
+  generateRandomQuote() {
+    this.randomQuote = this.quotes[Math.floor(Math.random()*this.quotes.length)]
+  }
+
+  getQuotesData(): void {
+    this.quotesService.getQuotes()
+      .subscribe(quotes => {
+        this.quotes = quotes.slice(1, 100);
+        this.getAuthors();
+        this.generateRandomQuote();
+        this.countPages();
+      });
+  }
+
+  countPages(): void {
+    let quotes = !this.quotesFiltered.length ? this.quotes : this.quotesFiltered;
+    this.pagesArray = quotes.length > this.quotesPerPage ? ([...Array(Math.ceil(quotes.length / 9)).keys()]) : [];
```

Comment:
Q: Does `9` have to be hardcoded? Or can it be replaced with a variable in case we wish to change it later?

Thanks! 🙏 🙇

Reply:
Yes, sorry. I have a variable for this value but forgot to add it here. 🙈
Thank you!🙏 🙇

Reply:
Awesome, thank you! 🙏

Reply:
Thank you! 🙏😊